src/libcore/num/mod.rs

```diff
@@ -2037,6 +2037,118 @@ assert_eq!((-a).rem_euclid(-b), 1);
             }
         }
 
+        doc_comment! {
+            concat!("Returns the logarithm of the number with respect to an arbitrary base.
+
+Returns `None` if the number is lower than 1.
+
+This method may not be optimized owing to implementation details;
+`self.log2()` can produce results more efficiently for base 2, and
+`self.log10()` can produce results more efficiently for base 10.
```

Comment:
`log10` is implemented as `log(10)`.

Reply:
the idea is that there will be a more efficient implementation for `log10` in the future.

for example:
```rust
impl u8 {
    fn log10(self) -> Option<u32> {
        match self {
            0 => None,
            1..=9 >= Some(0),
            10..=99 >= Some(1),
            100..=255 >= Some(2),
        }
    }
}
```